### PR TITLE
Upgrade React Query to v5

### PR DESCRIPTION
**Pull request by jbouder**
## Reference Issues or PRs

N/A

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Any other comments?
- Updated `@tanstack/react-query` to latest
- Updated `axios` to latest to resolve security vulnerability
- Updated server types page with proper query declaration for v5
- Fixed unit tests
- Included new build

### CODE REVIEW

File: ui/src/pages/server-types/server-types.test.tsx [hunk 1]
```diff
@@ -39,7 +41,7 @@ describe('ServerTypes', () => {
 
   // Error state test
   test('renders a message when no servers', () => {
-    queryClient.setQueryData(['serverTypes'], null);
+    queryClient.setQueryData(['server-types'], null);
```
Comment (aktech) [inline ui/src/pages/server-types/server-types.test.tsx:44]: Is this what's changed in the React Query v5?

File: ui/src/pages/server-types/server-types.tsx [hunk 2]
```diff
@@ -50,9 +50,18 @@ export const ServerTypes = (): React.ReactElement => {
     data: serverTypes,
     isLoading,
     error,
-  } = useQuery(['serverTypes'], async () => {
-    const { data } = await axios.get('/spawner-profiles/');
-    return data;
+  } = useQuery<AppProfileProps[], { message: string }>({
+    queryKey: ['server-types'],
```
Comment (aktech) [inline ui/src/pages/server-types/server-types.tsx:54]: I see, you changed it here! Is this related to upgrade of React Query?

Comment (jbouder) [reply ui/src/pages/server-types/server-types.tsx:54]: Yup, they changed how you have to use the hook

**Review by aktech — APPROVED** [review]
Seems like since the time you created this PR, they released another version: [5.28.6](https://www.npmjs.com/package/@tanstack/react-query/v/5.28.6)

Not expecting to upgrade in this PR, so I'll approve this.
